src/adapter/threads.ts

```diff
@@ -403,7 +402,7 @@ export class Thread implements IVariableStoreDelegate {
     }
   }
 
-  private _initialize() {
```

Comment:
Delaying attachment of these event listeners seems dangerous to me. E.g. is there a case where the user has a debugger statement in their script which gets hit almost immediately and we don't listening for the `paused` event in time?

Reply:
Those events should be send when we send `Debugger.enable` and not before.

Reply:
Well, if it hits a debugger statement before `Debugger.enable` will it still pause? I guess I don't know for sure.

Reply:
If we want to do that ASAP how about we make `this.dap` a deferred which will be resolved on 'initialize'? Then we can hook up handlers and they can send events as normal and they will effectively be queued and released after initialize.

Reply:
Won't that be caught by the fact that we start off at about:blank, send all the config (such as Debugger.enable) and _then_ navigate to the user's actual page?

Not as confident about what happens on the Node side of things, but I believe we always start paused there anyway right?

Reply:
Yea, V8 won't pause unless there's a debugger attached and enabled. I think you can simulate races pretty well by putting `await delay(1000)` at the top of `initialize`. For me this does cause us to miss breakpoints early on in the execution.